### ISSUE TITLE
Record pattern translation fails in mutual block?

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#2308]
This checks in Agda-2.5.1.1 but not in current stable-2.5:
```agda
data Unit : Set where
  unit : Unit

mutual -- needed

  record R : Set where  -- needs to be in mutual block
    constructor wrap
    field f : Unit

  data D : Set where
    c : R → D

  F : D → Set
  F (c (wrap _)) = Unit  -- match on wrap needed

test : (x : D) → F x → Set₁
test (c _) unit = Set
-- Type mismatch
-- when checking that the pattern unit has type F (c x)
```

**Comment by andreasabel** [comment]
Reduced case:

``` agda

data Unit : Set where
  unit : Unit

mutual -- needed

  record R : Set where  -- needs to be in mutual block
    constructor wrap
    field f : Unit

  G : R → Set
  G (wrap _) = Unit -- match on wrap needed

test : (x : R) → G x → Set₁
test _ unit = Set

-- Type mismatch
-- when checking that the pattern unit has type G x
```

**Comment by andreasabel** [comment]
Likely cause: https://github.com/agda/agda/commit/bb4e5b79e59a0cc24cfb723ef808214695f6c8cd to fix #2197. This also introduced #2209.
Eta-equality for records is only turned on after positivity checking which happens after the mutual block.  But the record pattern translation happens directly after the body of G is type-checked, thus, inside the mutual block.
